src/backend/distributed/operations/shard_transfer.c

```diff
@@ -390,6 +406,46 @@ citus_move_shard_placement(PG_FUNCTION_ARGS)
 }
 
 
+/*
+ * IsShardListOnNode determines whether a co-located shard list has
+ * active placements on a given node.
+ */
+static bool
+IsShardListOnNode(List *colocatedShardList, char *targetNodeName, uint32 targetNodePort)
+{
+	WorkerNode *workerNode = FindWorkerNode(targetNodeName, targetNodePort);
+	if (workerNode == NULL)
+	{
+		ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+						errmsg("Moving shards to a non-existing node is not supported")));
+	}
+
+	bool shardListIsPresent = false;
```

Comment:
It's not super clear to me what edge case this is handeling? 

Is this indirectly checking if the colocatedShardList is empty? If that's the case I'd prefer a simple return false with an if for clarity. But I would also like a comment, because it's not really clear to me why don't want to return true when the colocatedShardList is empty?